src/hooks/useFormJson.tsx

```diff
@@ -108,14 +108,17 @@ function validateFormsArgs(formUuid: string, rawFormJson: any): Error {
  * @param {string} [formSessionIntent] - The optional form session intent.
  * @returns {FormSchema} - The refined form JSON object of type FormSchema.
  */
-function refineFormJson(
+async function refineFormJson(
   formJson: any,
   schemaTransformers: FormSchemaTransformer[] = [],
   formSessionIntent?: string,
-): FormSchema {
+): Promise<FormSchema> {
```

Comment:
I have reservations to the Promise additions to this file because this affects the way the forms load. Maybe you can share why we need this

Reply:
This adjustment is needed because some fields need additional attributes that come from endpoints, and based on the response, we decide what to render. also this can also be done the transformer.